nibabel/freesurfer/tests/test_io.py

```diff
@@ -92,6 +94,32 @@ def test_morph_data():
     curv = read_morph_data(curv_path)
     assert_true(-1.0 < curv.min() < 0)
     assert_true(0 < curv.max() < 1.0)
+    with InTemporaryDirectory():
+        new_path = 'test'
+        write_morph_data(new_path, curv)
+        curv2 = read_morph_data(new_path)
+        assert_equal(curv2, curv)
+
+
+def test_write_morph_data():
+    """Test write_morph_data edge cases"""
+    values = np.arange(20, dtype='>f4')
+    okay_shapes = [(20,), (20, 1), (20, 1, 1), (1, 20)]
+    bad_shapes = [(10, 2), (1, 1, 20, 1, 1)]
+    big_num = np.iinfo('i4').max + 1
+    with InTemporaryDirectory():
+        for shape in okay_shapes:
+            write_morph_data('test.curv', values.reshape(shape))
+            # Check ordering is preserved, regardless of shape
+            assert_equal(values, read_morph_data('test.curv'))
+        assert_raises(ValueError, write_morph_data, 'test.curv',
+                      np.zeros(shape), big_num)
+        if ctypes.c_long is not ctypes.c_int32:
+            assert_raises(ValueError, write_morph_data, 'test.curv',
+                          strided_scalar((big_num,)))
```

Comment:
Doesn't seem to capture the case for [PYTHON_VERSION=2.7, PYTHON_ARCH=32](https://ci.appveyor.com/project/Eric89GXL/nibabel/build/1.0.589).